microsoft-edge/devtools-guide-chromium/whats-new/2024/08/devtools-128.md

```diff
@@ -45,6 +48,36 @@ The top of each context menu now shows the title again:
 ![New context menu header](./devtools-128-images/header-title.png)
 
 
+<!-- ====================================================================== -->
+## Heap snapshot improvements
+
+<!-- Subtitle: Heap snapshots are generated faster, HTML elements are grouped by tag name, shallow sizes show fewer internal objects, and numeric values are always shown. -->
+
+In the **Memory** tool, there are several improvements for heap snapshots:
+
+* Generating heap snapshots is now faster.
+
+* HTML elements are grouped by tag name.<!-- todo: by element name? -->
+
+* Shallow sizes now match JavaScript semantics better, by showing fewer internal objects, unless the **Expose internals (includes additional implementation-specific details)** setting is enabled<!--todo: checkbox is selected?  where/what is this UI control?  not found after turned experiment on, and then created snapshot --> when the [Show option to expose internals in heap snapshots](../../../experimental-features/index.md#show-option-to-expose-internals-in-heap-snapshots) experiment is turned on.
```

Comment:
"unless the **Expose internals (includes additional implementation-specific details)** setting** is enabled ...

Does that mean when the **Expose internals (includes additional implementation-specific details)** checkbox is selected?  where/what is this UI control?  such a UI control is not found after turned the experiment on, and then created snapshot.
@leahmsft

Reply:
Yes, it is referring to the checkbox below Heap snapshot (before you take the snapshot): 
![image](https://github.com/user-attachments/assets/bbc0eb8e-ae65-4418-91f6-8122af76d8fa)

Reply:
result:
```
Shallow sizes now match JavaScript semantics better, by showing 
fewer internal objects (unless you select 
the [Show option to expose internals in heap snapshots](../../../experimental-features/index.md#show-option-to-expose-internals-in-heap-snapshots) 
experiment checkbox and then, before taking a snapshot, select 
the **Expose internals (includes additional implementation-specific details)** 
checkbox).
```